server/game/cards/01_SOR/units/RelentlessKonstantinesFolly.ts

```diff
@@ -0,0 +1,42 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { CardType, RelativePlayer, Location } from '../../../core/Constants';
+import { StateWatcherRegistrar } from '../../../core/stateWatcher/StateWatcherRegistrar';
+import { CardsPlayedThisPhaseWatcher } from '../../../stateWatchers/CardsPlayedThisPhaseWatcher';
+
+export default class RelentlessKonstantinesFolly extends NonLeaderUnitCard {
+    private cardsPlayedThisPhaseWatcher: CardsPlayedThisPhaseWatcher;
+
+    protected override getImplementationId() {
+        return {
+            id: '3401690666',
+            internalName: 'relentless#konstantines-folly'
+        };
+    }
+
+    protected override setupStateWatchers(registrar: StateWatcherRegistrar): void {
+        this.cardsPlayedThisPhaseWatcher = AbilityHelper.stateWatchers.cardsPlayedThisPhase(registrar, this);
+    }
+
+    public override setupCardAbilities() {
+        this.addConstantAbility({
+            title: 'The first event played by each opponent each round loses all abilities',
+            ongoingEffect: AbilityHelper.ongoingEffects.blank(false),
+            //An event card goes to the discard before its effects resolve, so filtering for being in the discard is appropriate to allow playing an event but blanking its effects.
```

Comment:
I was confused why this is the case, since the "play event" ability should not be getting blanked by the change you made. I investigated and there's a bug related to some old blanking code ported from l5r. Take a look at this commit with the bug fix, it passes your tests: https://github.com/AMMayberry1/forceteki/commit/fbcad5f10ea8a4043d17a36d8755c5fe1dddb05a

I would prefer we use that version since it's clearer and simpler (and it fixes the bug).

Reply:
Sure. I've switched it over.